server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -220,6 +220,8 @@ Metadata upgradeMetadataForNode(Metadata metadata,
     public static Metadata upgradeMetadata(Metadata metadata,
                                     MetadataIndexUpgradeService metadataIndexUpgradeService,
                                     MetadataUpgrader metadataUpgrader) {
+        metadata = metadataUpgrader.indexTemplateVersionCreatedFixer.apply(metadata);
+
```

Comment:
Not sure upgrading here is enough, see https://github.com/crate/crate/pull/17175.

If we go with this approach, we probably should just add upgrade logic to the `applyPluginUpgraders` since it's used in all places where we want to upgrade templates.